src/clj/rems/api/schema.clj

```diff
@@ -196,7 +196,8 @@
 (s/defschema ApplicationAttachment
   {:attachment/id s/Int
    :attachment/filename s/Str
-   :attachment/type s/Str})
+   :attachment/type s/Str
+   :attachment/occurs-in (s/maybe #{(s/enum :event :value :previous-value)})})
```

Comment:
Would it make sense to also store the event ids where the attachment occurred?

Reply:
Perhaps, but that would make for a more complicated schema. Besides, you won't know what the event id number means without walking all the events, at which point you can just check where the attachment occurs anyway. I think the usecase for this is something like "grab all :value attachments and upload them to our CRM"

Reply:
Do we actually know enough about the use cases to do any concrete API design? Do we need to expose this information in the API? Can't we do the download as zip even without this?

Reply:
Yeah, we could only keep this internally, but there was an actual request for this, see #2571. I guess some integrations want to duplicate the "all attachments" functionality without going via a zip?

Reply:
Yes but with that level of specification I'm just afraid we'll miss and end up an API that keeps breaking. Like "attachment to a decision" could mean a lot of things and this does not help to answer that.

Reply:
Yeah we discussed the granularity of classifications with Jaakko on flowdock and ended up with what's in this PR now. I don't think it makes sense to make this more fine-grained. At that point the users could just walk the application structure anyway.

Reply:
Yeah, I was just saying that perhaps we can skip the public API altogether. The issues we have don't require it in my mind. The integrators could actually use examples in the docs?